src/services/balancer/contracts/contracts/vault.ts

```diff
@@ -112,17 +112,34 @@ export default class Vault {
     vaultMultiCaller.call('poolTokens', this.address, 'getPoolTokens', [id]);
 
     if (isStablePhantom(type) && result.linearPools) {
-      // Get pool tokens for linear pools
+      const wrappedTokensMap: Record<string, string> = {};
+
       Object.keys(result.linearPools).forEach(address => {
         if (!result.linearPools) return;
         const linearPool: RawLinearPoolData = result.linearPools[address];
+
         vaultMultiCaller.call(
           `linearPools.${address}.tokenData`,
           this.address,
           'getPoolTokens',
           [linearPool.id]
         );
+
+        wrappedTokensMap[address] = linearPool.wrappedToken.address;
+      });
+
+      Object.entries(wrappedTokensMap).forEach(([address, wrappedToken]) => {
+        poolMulticaller.call(
+          `${address}.unwrappedTokenAddress`,
+          wrappedToken,
+          'ATOKEN'
+        );
       });
+
+      const callResult = await poolMulticaller.execute();
+      if (callResult != null) {
+        merge(result.linearPools, callResult);
```

Comment:
@garethfuller WDYT about this way of merging the results? They both share the same `key` (the address), so I think its safe to do it.

Reply:
Can you not just modify the `result` like we're doing elsewhere? I think execute already handles this kind of merging of data.
```ts
Object.entries(wrappedTokensMap).forEach(([address, wrappedToken]) => {
  poolMulticaller.call(
    `linearPools.${address}.unwrappedTokenAddress`,
    wrappedToken,
    'ATOKEN'
  );
});

const result = await poolMulticaller.execute(result);
```

Reply:
Thanks, much cleaner :)